x-pack/plugins/security_solution/public/timelines/components/flyout/header/kpis.tsx

```diff
@@ -5,61 +5,95 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useMemo } from 'react';
 
-import { EuiStat, EuiFlexItem, EuiFlexGroup } from '@elastic/eui';
+import { EuiStat, EuiFlexItem, EuiFlexGroup, EuiToolTip } from '@elastic/eui';
+import numeral from '@elastic/numeral';
+import { DEFAULT_NUMBER_FORMAT } from '../../../../../common/constants';
+import { useUiSetting$ } from '../../../../common/lib/kibana';
 import { TimelineKpiStrategyResponse } from '../../../../../common/search_strategy';
 import { getEmptyValue } from '../../../../common/components/empty_value';
 import * as i18n from './translations';
 
 export const TimelineKPIs = React.memo(
   ({ kpis, isLoading }: { kpis: TimelineKpiStrategyResponse | null; isLoading: boolean }) => {
+    const kpiFormat = '0,0.[000]a';
+    const [defaultNumberFormat] = useUiSetting$<string>(DEFAULT_NUMBER_FORMAT);
+    const formattedKpis = useMemo(() => {
+      return {
+        process: kpis === null ? getEmptyValue() : numeral(kpis.processCount).format(kpiFormat),
+        user: kpis === null ? getEmptyValue() : numeral(kpis.userCount).format(kpiFormat),
+        host: kpis === null ? getEmptyValue() : numeral(kpis.hostCount).format(kpiFormat),
+        sourceIp: kpis === null ? getEmptyValue() : numeral(kpis.sourceIpCount).format(kpiFormat),
+        destinationIp:
+          kpis === null ? getEmptyValue() : numeral(kpis.destinationIpCount).format(kpiFormat),
+      };
+    }, [kpis]);
+    const formattedKpiToolTips = useMemo(() => {
+      return {
+        process: numeral(kpis?.processCount).format(defaultNumberFormat),
```

Comment:
same here, if the field is undefined for whatever reason how will it be formatted?

Reply:
Will show 0 in that case.